test/hotspot/jtreg/compiler/lib/ir_framework/test/AbstractTest.java

```diff
@@ -109,7 +109,21 @@ public void run() {
 
     abstract protected void compileTest();
 
+    /**
+     * Exception thrown when a test method is compiled that is not compilable, and this is
+     * explicitly allowed with "allowNotCompilable".
+     */
+    private class MethodNotCompilableException extends Exception {}
+
     protected void compileMethod(DeclaredTest test) {
+        try {
+            tryCompileMethod(test);
+        } catch (MethodNotCompilableException e) {
+            TestRun.check(test.isAllowNotCompilable(), "Only allowNotCompilable methods should throw MethodNotCompilableException.");
```

Comment:
Should we emit a log message here in case we have an expected compilation bailout here?

Reply:
done!